x/evm/keeper/grpc_query.go

```diff
@@ -254,12 +254,17 @@ func (k Keeper) EthCall(c context.Context, req *txs.EthCallRequest) (*txs.MsgEth
 	}
 
 	txConfig := states.NewEmptyTxConfig(common.BytesToHash(ctx.HeaderHash()))
+	// use cache ctx and ignore the write function, since we don't need to commit the stateDB
+	cacheCtx, _ := ctx.CacheContext()
```

Comment:
Seems we don't have to worry about writting issue, ctx is create in DoCall, ContextWithHeight. And we don't commit to state if the tx comes from an local call.

Reply:
Fixed